includes/class-swwrc-uc-taxonomy-terms-widget.php

```diff
@@ -0,0 +1,153 @@
+<?php
+
+class SWWRC_UC_Taxonomy_Terms_Widget extends WP_Widget {
+
+	/**
+	 * Register the widget officially through the parent class.
+	 *
+	 * @since 0.5.0
+	 */
+	public function __construct() {
+		parent::__construct(
+			'uc_taxonomies_widget',
+			'University Center Taxonomy Terms',
+			array(
+				'description' => 'Display the taxonomy terms used for the selected University Center post type.',
+			)
+		);
+	}
+
+	/**
+	 * Outputs the content of the widget.
+	 *
+	 * @since 0.5.0
+	 *
+	 * @param array $args
+	 * @param array $instance
+	 */
+	public function widget( $args, $instance ) {
+		$default_instance = array(
+			'title' => '',
+			'post_type' => '',
+			'taxonomy' => '',
+		);
+
+		$instance = shortcode_atts( $default_instance, $instance );
+		$post_types = array_merge( wsuwp_uc_get_object_type_slugs(), array( 'post' ) );
+
+		if ( empty( $instance['post_type'] ) || empty( $instance['taxonomy'] ) ) {
+			return '';
+		}
+
+		if ( ! in_array( $instance['post_type'], $post_types, true ) ) {
+			return '';
+		}
+
+		if ( ! in_array( $instance['taxonomy'], array( 'post_tag', 'category' ), true ) ) {
+			return '';
+		}
+
+		$terms = get_terms( $instance['taxonomy'], array(
+			'post_type' => $instance['post_type'],
+		) );
+
+		if ( empty( $terms ) && is_wp_error( $terms ) ) {
+			return '';
+		}
+
+		$home = trailingslashit( get_home_url() );
+		$taxonomy_slug = get_taxonomy( $instance['taxonomy'] )->rewrite['slug'];
+
+		if ( 'post' === $instance['post_type'] ) {
+			$url_base = $home . $taxonomy_slug . '/';
```

Comment:
Instead of storing `$home` locally above, I think this this could be `$url_base = get_home_url( null, $taxonomy_slug . '/' );` and similar just below.

It's also fine as is :)